src/lit/video.ts

```diff
@@ -0,0 +1,209 @@
+import * as dom from '../dom/dom';
+import * as func from '../func/func';
+
+import {
+  ElementVisibilityObject,
+  elementVisibility,
+} from '../dom/element-visibility';
+import {LitElement, html} from 'lit';
+import {property} from 'lit/decorators.js';
+
+import {DomWatcher} from '../dom/dom-watcher';
+
+/**
+ * # Degu Video Component
+ *
+ * Usage:
+ *
+ * ```ts
+ * import { DeguVideo} from '@blinkk/degu/lib/lit/video';
+ *
+ * window.customElements.define('degu-video', DeguVideo);
+ * ```
+ *
+ * ```html
+ *   <degu-video
+ *     src="https://storage.googleapis.com/mannequin/blobs/7b0c13d7-e9cc-4efa-93c1-9db323d7283b.mp4"
+ *     width="640"
+ *     height="640"
+ *     style="aspect-ratio: 1"
+ *     a11y-label="Video Aria Label"
+ *     autoplayinview="true"
+ *   ></degu-video>
+ * ```
+ *
+ * ```css
+ *
+ *   degu-video {
+ *     display: inline-block;
+ *     line-height: 0;
+ *     width: 100%
+ *   }
+ *
+ *   degu-video video {
+ *       width: 100%;
+ *   }
+ * ```
+ */
+export class DeguVideo extends LitElement {
+  @property({type: String, attribute: 'src'})
+  src: string;
+
+  @property({type: String, attribute: 'a11y-label'})
+  private a11yLabel: string;
```

Comment:
Let's call this `aria-label` to be consistent with "extending the default element"?

Reply:
Some concerns with this change.

So right now, it renders as follows

```html
<degu-video a11y-label="Video Aria Label" >
      <video role="img" aria-label="Video Aria Label" load="true">
         ...
      </video>
    </degu-video>
```

This gets announced as "Video Aria Label : Image" (given the role="img").

The change will modify this.  We end up with two aria labels, which gets announced as a single "Video Aria Label: Group" (in VO) - VO doesn't seem to focus into the video.

```html
<degu-video aria-label="Video Aria Label" >
      <video role="img" aria-label="Video Aria Label" load="true">
         ...
      </video>
    </degu-video>
```

We could "remove" the top level aria-label from degu-video but this would also have to make it a non-reactive property.

Reply:
Ah right, sorry, I forgot about this concern. OK - another alternative. What if we just added `role=img` to the host element? Then `aria-label` could be left unmodified.

Reply:
FWIW I think we should come up with some convention that will work across all custom elements while preserving as much of the native experience as we can.

Downside of applying `role=img` to the custom element is that we'll probably need to do something like `<img aria-hidden=true>` to avoid duplicate reading.

Need to also consider something like `<custom-button>` which would have an embedded button, we shouldn't really add `role="button"` to that one.

Reply:
Video is a bit unique because the builtin element doesn't have consistent screen reader support, so the approach we want to take with it is:

```<video role="img" aria-label="Aria label">```

I think if we approach building our custom elements as "enhancing the functionality of builtin elements" we should try to stay aligned with the built-in attributes, roles, etc. -- so I think it'll have to be done on a case-by-case basis depending on the element?

For video specifically I think when we checked this last, the videos weren't announced. But yes, we'd need to add `<video aria-hidden="true">` to enforce that.

Reply:
Okay updated and also added some test.  I tested it on VO and it's working fine.